### PR TITLE
HSEARCH-4231 Static sharding of automatic indexing outbox event queues

### PR DESCRIPTION
**Pull request by yrodiere**
https://hibernate.atlassian.net/browse/HSEARCH-4231

### CODE REVIEW

File: integrationtest/mapper/orm/src/test/java/org/hibernate/search/integrationtest/mapper/orm/automaticindexing/strategy/outbox/OutboxPollingStaticShardingBaseIT.java [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+				}
+			} );
+		}
+		backendMock.verifyExpectationsMet();
+		// All works must be executed exactly once
+		assertIndexingCountAcrossAllSessionFactories().isEqualTo( entityCount );
+		// The workload must be spread uniformly (with some tolerance)
+		assertIndexingCountForEachSessionFactory()
+				.allSatisfy( count -> assertThat( count )
+						.isCloseTo( entityCount / totalShardCount, Percentage.withPercentage( 25 ) ) );
```
Comment (fax4ever) [inline integrationtest/mapper/orm/src/test/java/org/hibernate/search/integrationtest/mapper/orm/automaticindexing/strategy/outbox/OutboxPollingStaticShardingBaseIT.java:241]: very nice to check that!

**Review by fax4ever — APPROVED** [review]
Great work!
+1 to merge it

**Comment by yrodiere** [comment]
Thanks for the review @fax4ever . I addressed your comments; I will merge as soon as the build passes.

**Comment by fax4ever** [comment]
Thank you! Great job

**Comment by fax4ever** [comment]
Merged, thanks!

**Pull request merged**